client/webserver/site/src/js/markets.ts

```diff
@@ -2590,12 +2673,19 @@ export default class MarketsPage extends BasePage {
       page.lotField.value = '0'
       page.qtyField.value = ''
       this.previewQuoteAmt(false)
+      this.setOrderBttnEnabled(false, intl.prep(intl.ID_ORDER_BUTTON_QTY_ERROR))
       return
     }
     const lotSize = this.market.cfg.lotsize
+    const orderQty = lots * lotSize
     page.lotField.value = String(lots)
     // Conversion factor must be a multiple of 10.
-    page.qtyField.value = String(lots * lotSize / this.market.baseUnitInfo.conventional.conversionFactor)
+    page.qtyField.value = String(orderQty / this.market.baseUnitInfo.conventional.conversionFactor)
+
+    if (!this.isLimit() && this.isSell()) {
+      const baseWallet = app().assets[this.market.base.id].wallet
+      this.setOrderBttnEnabled(orderQty <= baseWallet.balance.available, intl.prep(intl.ID_ORDER_BUTTON_SELL_BALANCE_ERROR))
+    }
```

Comment:
You're only setting the button status for market sells here? Why not limit orders?

Reply:
Generally the order button state is managed from `preBuy()` and `preSell()`.  The market buy order logic is different and those methods don't get called so that case has to be handled here.

Reply:
Please leave a good comment then. My guess is we can handle this somewhere more natural though.

Reply:
I'm having trouble seeing where market sells are treated differently such that the `updateOrderBttnState` isn't called.

Reply:
It's the market buy orders that don't have the lot field - sorry about the confusion above.  Adding a comment.

Reply:
That doesn't really help me. This little block only runs for market sells. Why? You said

> The market sell order logic is different and those methods don't get called so that case has to be handled here.

Where is the market sell order logic handled different such that the order button state is not updated.

Reply:
`previewQuoteAmt(true)` doesn't call `preSell` which calls `updateOrderBttnState` to update the button, hence the special case here.

https://github.com/peterzen/dcrdex/blob/204a7d0f03221859cb6170b847ab759ddfb2317f/client/webserver/site/src/js/markets.ts#L1334-L1338

    if (!show || !adjusted || !order.qty) {
      page.orderPreview.textContent = ''
      this.drawChartLines()
      return
    }

Reply:
Oh. It's because `adjusted` in `NaN`.